api/src/main/java/jakarta/persistence/EntityAgent.java

```diff
@@ -0,0 +1,266 @@
+     *         determined to not be detached
+     */
+    void delete(Object entity);
+
+    /**
+     * Delete every record in the given list. The records are
+     * deleted in the order in which they occur in the given list.
+     *
+     * @param entities The entities to be deleted.
+     *
+     * @see #delete(Object)
+     */
+    void deleteMultiple(List<?> entities);
+
+    /**
+     * Perform an upsert, that is, insert the record if it does not
+     * exist, or update it if it already exists.
+     * <p>
+     * This method never performs id generation and does not accept
+     * an entity instance with a null identifier. When id generation
+     * is required, use {@link #insert(Object)}.
+     * <p>
+     * On the other hand, {@code upsert()} does accept an entity
+     * instance with an assigned identifier value, even if the entity
+     * {@link Id} field is declared to be generated, for example, if
+     * it is annotated {@link GeneratedValue}. Thus, this method may
+     * be used to import data from an external source.
+     * <p>
+     * <ul>
+     * <li>The {@link jakarta.persistence.PreUpsert} callback is
+     *     triggered.
+     * <li>The {@link jakarta.persistence.PostUpsert} callback is
+     *     triggered if the operation is successful.
+     * </ul>
+     *
+     * @param entity a detached entity instance, or a new instance
+     *               with an assigned identifier
+     * @throws IllegalArgumentException if the entity has a null id
+     */
+    void upsert(Object entity);
+
+    /**
+     * Upsert every record in the given list. The records are
+     * upserted in the order in which they occur in the given list.
+     *
+     * @param entities The entities to be inserted or updated.
+     *
+     * @see #upsert(Object)
+     */
+    void upsertMultiple(List<?> entities);
+
+    /**
+     * Retrieve a record.
+     *
+     * @param entityClass The class of the entity to retrieve
+     * @param id The id of the entity to retrieve
+     *
+     * @return a detached entity instance
+     */
+    <T> T get(Class<T> entityClass, Object id);
+
+    /**
+     * Retrieve a record, obtaining the specified lock mode.
+     *
+     * @param entityClass The class of the entity to retrieve
+     * @param id The id of the entity to retrieve
+     * @param lockMode The lock mode to apply to the entity
+     *
+     * @return a detached entity instance
+     */
+    <T> T get(Class<T> entityClass, Object id, LockModeType lockMode);
+
+    /**
+     * Retrieve a record, fetching associations specified by the
+     * given {@link EntityGraph}.
+     *
+     * @param graph The {@link EntityGraph}
+     * @param graphSemantic a {@link GraphSemantic} specifying
+     *                      how the graph should be interpreted
+     * @param id The id of the entity to retrieve
+     *
+     * @return a detached entity instance
+     *
+     * @since 6.3
+     */
+    <T> T get(EntityGraph<T> graph, Object id);
```

Comment:
```suggestion
     * @return a detached entity instance if a matching entity
     *             is found in the database. Otherwise, returns
     *             {@link Optional#empty()}
     */
    <T> Optional<T> get(EntityGraph<T> graph, Object id);
```